src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -176,10 +189,14 @@ EPOCHS = 2; // make test faster in Debug
       cout << "Epoch = " << e << endl;
       cout << "Anomaly = " << an << endl;
       cout << "Anomaly (avg) = " << avgAnom10.getCurrentAvg() << endl;
-      cout << "Anomaly (Likelihood) = " << anLikely << endl;
       cout << "SP (g)= " << outSP << endl;
       cout << "SP (l)= " << outSPlocal <<endl;
       cout << "TM= " << outTM << endl;
+      cout << "Cls[0]= " << categoryToReal_(argmax(pred.infer(outTM)[0])) << endl;
+      cout << "Cls[100]= " << categoryToReal_(argmax(pred.infer(outTM)[100])) << endl;
+
+      NTA_CHECK( categoryToReal_(argmax(pred.infer(outTM)[0])) != -1) << "Classifier did not learn"; //FIXME Predictor is not learning, this should be ~ sin(49.99)
```

Comment:
FIXME: this is always 0, predictor/classifier is not learning! 
Although the unit-tests are passing. 
@Thanh-Binh PR #667 should have fixed the symptomps you've described. But my predictor is still failing here. 
Is this the same issue as you described, does it replicate your (failing) experiments?

Reply:
@Thanh-Binh ping, can you please have a look at this PR and the test here? I'm getting 0s from the Classifier inference and not sure why.